### DISCUSSION TITLE
org.eclipse.egit.github.core.CommitComment / setPath(String path)

### DISCUSSION THREAD

**Discussion by Molama** [community/community#24873] [Apps, API and Webhooks]
<p>Heja,</p>
<p>I am trying to add an comment on a Pull Request from an Java Application in Eclipse using  <a href="https://github.com/eclipse/egit-github/tree/master/org.eclipse.egit.github.core" rel="nofollow noopener">GitHub Java API (org.eclipse.egit.github.core)</a>. </p>
<p>The scenario should be:</p>
<ol>
<li>recieving the payload from the webhook at the pull request</li>
<li>connect to the repository</li>
<li>get the pull request</li>
<li>add the comment (using PullRequestService.createComment(…); one of the Parameters is CommitComment -&gt; create Object and set necessary variables.</li>
</ol>
<p>But then i got this exception: </p>
<p>org.eclipse.egit.github.core.client.RequestException: Invalid request.</p>
<p>No subschema in “oneOf” matched.<br>
For ‘properties/commit_id’, nil is not a string.<br>
For ‘properties/path’, nil is not a string.</p>
<p>I don’t know what to set for commit_id and path? Can I get those information from the webhook-payload somehow?</p>
<p>Thank you!</p>

**Comment by lee-dohm** [comment, marked as answer]
<p>Commit comments, which you mention in the title of your post, are separate and different from Pull Request comments, which you mention in the body of your post. There are also Pull Request review comments, which are another type of comment associated with Pull Requests.</p>
<p>If you want to simply add a comment to the pull request conversation page that is not a review comment, you should use the <a href="https://developer.github.com/v3/issues/comments/#create-a-comment" rel="nofollow noopener">same endpoint as you would for creating a comment on an issue</a>. I don’t know what that translates to in your Java library, but that should point you in the right direction.</p>
<p>Let us know if you have more questions.</p>
